packages/typescript-plugin/src/type-tree/display-parts-builder.ts

```diff
@@ -0,0 +1,206 @@
+import * as ts from "typescript/lib/tsserverlibrary";
+
+const keywords = new Set([
+  "abstract",
+  "any",
+  "as",
+  "async",
+  "await",
+  "bigint",
+  "boolean",
+  "break",
+  "case",
+  "catch",
+  "class",
+  "const",
+  "continue",
+  "declare",
+  "default",
+  "do",
+  "else",
+  "enum",
+  "export",
+  "extends",
+  "false",
+  "finally",
+  "for",
+  "function",
+  "if",
+  "implements",
+  "import",
+  "in",
+  "infer",
+  "interface",
+  "is",
+  "keyof",
+  "let",
+  "module",
+  "namespace",
+  "never",
+  "new",
+  "null",
+  "number",
+  "object",
+  "out",
+  "private",
+  "protected",
+  "public",
+  "readonly",
+  "return",
+  "static",
+  "string",
+  "super",
+  "switch",
+  "symbol",
+  "this",
+  "throw",
+  "true",
+  "try",
+  "type",
+  "typeof",
+  "undefined",
+  "unique",
+  "unknown",
+  "var",
+  "void",
+  "while",
+  "yield",
+]);
+
+/**
+ * Helper utilities for creating SymbolDisplayPart objects with proper semantic classification.
+ * These functions simplify the creation of display parts during type tree generation.
+ */
+export const displayParts = {
```

Comment:
This should be all the available displayParts, according to the latest TS types.